server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java

```
@@ -0,0 +1,293 @@
+          .getFraction(Property.TABLE_MAX_MERGEABILITY_THRESHOLD);
+      if (mergeabilityThreshold <= 0) {
+        log.trace("Skipping FindMergeableRangeTask for table {}, {}} is set to {}", tableName,
+            Property.TABLE_MAX_MERGEABILITY_THRESHOLD.getKey(), mergeabilityThreshold);
+        continue;
+      }
+
+      long maxFileCount =
+          context.getTableConfiguration(tableId).getCount(Property.TABLE_MERGE_FILE_MAX);
+      long maxTotalSize = (long) (threshold * mergeabilityThreshold);
+
+      log.debug("Checking {} for tablets that can be merged", tableName);
+      log.trace("maxFileCount: {}, maxTotalSize:{}, splitThreshold:{}, mergeabilityThreshold:{}",
+          maxFileCount, maxTotalSize, threshold, mergeabilityThreshold);
+      try {
+        NamespaceId namespaceId = context.getNamespaceId(tableId);
+        var type = FateInstanceType.fromTableId(tableId);
+
+        try (var tablets = context.getAmple().readTablets().forTable(tableId)
+            .fetch(PREV_ROW, FILES, MERGEABILITY).filter(FILTER).build()) {
+
+          final MergeableRange current =
+              new MergeableRange(tableId, manager.getSteadyTime(), maxFileCount, maxTotalSize);
+
+          for (var tm : tablets) {
+            log.trace("Checking tablet {}, {}", tm.getExtent(), tm.getTabletMergeability());
+            // If there was an error adding the next tablet to the range then
+            // the existing range is complete as we can't add more tablets so
+            // submit a merge fate op and reset to find more merge ranges
+            current.add(tm).ifPresent(error -> {
+              submit(current, type, table, namespaceId);
+              current.resetAndAdd(tm);
+            });
+          }
+
+          // Try and submit any outstanding mergeable tablets
+          submit(current, type, table, namespaceId);
+        }
+
+      } catch (Exception e) {
+        log.error("Failed to generate system merges for {}", tableName, e);
+      }
```

@cshannon This completely swallows any exception, even new ones added to the code later. Are you sure that is what is intended here?

It seems to me that, at the very least, this could be narrowed to the current exceptions, so it doesn't catch unintended things that get added later:

```java
      } catch (TableNotFoundException | RuntimeException e) {
        log.error("Failed to generate system merges for {}", tableName, e);
      }
```

---

I pushed an update to main with this change.